sus/ptr/nonnull.h

```diff
@@ -197,7 +197,9 @@ struct fmt::formatter<::sus::ptr::NonNull<T>, Char> {
   template <class FormatContext>
   constexpr auto format(const ::sus::ptr::NonNull<T>& t,
                         FormatContext& ctx) const {
-    return underlying_.format(t.as_ptr(), ctx);
+    // t.as_ptr() returns `const T*`, so we need to cast its constness away in
+    // order to format it.
+    return underlying_.format(const_cast<T*>(t.as_ptr()), ctx);
```

Comment:
Should the `underlying_` formatter be a const void* instead?

Reply:
Ah, that works! I tried with `const volatile void*`, and didn't think to try just `const void*` 😳